apps/internal/base/storage/items.go

```diff
@@ -83,7 +84,7 @@ type AccessToken struct {
 }
 
 // NewAccessToken is the constructor for AccessToken.
-func NewAccessToken(homeID, env, realm, clientID string, cachedAt, expiresOn, extendedExpiresOn time.Time, scopes, token, tokenType, authnSchemeKeyID string) AccessToken {
+func NewAccessToken(homeID, env, realm, clientID string, cachedAt, expiresOn, extendedExpiresOn time.Time, scopes, token, tokenType, authnSchemeKeyID string, resource string) AccessToken {
```

Comment:
Do we really need resource here? Can you reuse scopes instead? From caching point of view, it's just a string, we don't need to differentiate. The logical layer (the MSI classes) will know how to deal with it.

Reply:
Removed.